matrix-client-handlers.el

```diff
@@ -123,8 +124,7 @@ like."
    (own-display-name (oref* room :con :username)))
 
   ((pcase-let (((map event_id) data)
-               (metadata)
-               (output))
+               (metadata) (message) (matrix-image-url))
```

Comment:
> By setting (message) to nil here (I think that's what is happening in this pcase, right? I'm pretty new to pcase)

Yes, `pcase-let` is new to me too.  Unlike `let`, it doesn't support plain symbols to bind to nil, so they must be specified in a single-element list.  In a future commit I will be simplifying this slightly by getting rid of the `message` bind, since it's only used in one place.  Anyway...

Good catch here!  So I guess this means that redacted messages are not actually removed from the event graph, just their contents are redacted, and the redaction is added to the unsigned object.

Well, this will be easy to fix: I just need to add a case to `(pcase msg-type` for when it's `nil`, and then do nothing if there is no message to insert.

Glad you caught this!

Reply:
Fixed in separate commit.